src/common/pf_snmp.c

```diff
@@ -225,25 +236,123 @@ static void pf_snmp_encode_signal_delays (
    encoded->line_propagation_delay_ns = plain->cable_delay_local;
 }
 
-void pf_snmp_get_system_name (pnet_t * net, pf_snmp_system_name_t * name)
+void pf_snmp_data_init (pnet_t * net)
 {
    const char * directory = pf_cmina_get_file_directory (net);
    int error;
    int result;
 
-   CC_STATIC_ASSERT (sizeof (name->string) >= PNAL_HOSTNAME_MAX_SIZE);
+   CC_STATIC_ASSERT (
+      sizeof (net->snmp_data.system_name.string) >= PNAL_HOSTNAME_MAX_SIZE);
 
-   error = pf_file_load (directory, PF_FILENAME_SYSNAME, name, sizeof (*name));
+   /* sysContact */
+   error = pf_file_load (
+      directory,
+      PF_FILENAME_SNMP_SYSCONTACT,
+      &net->snmp_data.system_contact,
```

Comment:
Suggestion: use a temporary variable to reduce repetition, e.g snmp_data = &net->snmp_data. This makes the code somewhat easier to follow.

Reply:
Introduced
`pf_snmp_data_t * snmp = &net->snmp_data;`
for` pf_snmp_data_init()` and `pf_snmp_data_clear()` 
Done